packages/test/test-version-utils/src/versionUtils.ts

```diff
@@ -430,7 +406,62 @@ export function getRequestedVersion(
 		// cap at min version
 		return "^0.0.1-0";
 	}
-	return resolveVersion(`^0.${requestedMinorVersion}.0-0`, false);
+	return `^0.${requestedMinorVersion}.0-0`;
+}
+
+/**
+ *
+ * Given a version, returns the most recently released version. The version provided can be adjusted to
+ * the next or previous major versions by providing positive/negative integers in the `requested` parameter.
+ *
+ * @param baseVersion - The base version to move from (eg. "0.60.0")
+ * @param requested - If the value is a negative number, the baseVersion will be adjusted down.
+ * If the value is a string then it will be returned as-is. Throws on positive number.
+ * @param adjustPublicMajor - If `baseVersion` is a Fluid internal version, then this boolean controls whether the
+ * public or internal version is adjusted by the `requested` value. This parameter has no effect if `requested` is a
+ * string value or if `baseVersion` is not a Fluid internal version.
+ *
+ * @remarks
+ *
+ * In typical use, the `requested` values are negative values to return ranges for previous versions (e.g. "-1").
+ *
+ * @example
+ * ```typescript
+ * const newVersion = getRequestedVersion("2.3.5", -1); // "^1.0.0"
+ * ```
+ *
+ * @example
+ * ```typescript
+ * const newVersion = getRequestedVersion("2.3.5", -2); // "^0.59.0"
+ * ```
+ *
+ * @internal
+ */
+export function getRequestedVersion(
```

Comment:
@alexvy86, I was also thinking about this too. I don't think it should be very risky, because we shouldn't be hitting the fail-retry logic outside of this niche case. Like you mentioned, I don't think we really have any cases where the `-2` retry would trigger and we don't want it.

Also, this may be off topic, but I think we can argue that this is closer to what the expected behavior for `getRequestedVersion()` should be in general. I think when we refactor this code, we will want to change the language on N-X a bit, especially now that we have internal, RC and public releases. The language should probably be something like "if you request N-X, you should get the Xth release before N", rather than subtracting X from N's version mathematically (we kind of do this already when requesting N-X for RC1).